app/controllers/LandingController.scala

```diff
@@ -213,19 +212,18 @@ class LandingController @Inject()(val landingService: LandingService,
       case Some(amlsRegistrationNumber) => landingService.cacheMap flatMap {
         //enrolment exists
         case Some(c) =>
-          val fix = for {
-            c2 <- fixEmptyRecords[ResponsiblePeople](c, ResponsiblePeople.key)
+          lazy val fixEmpties = for {
+            c1 <- fixEmptyRecords[TradingPremises](c, TradingPremises.key)
+            c2 <- fixEmptyRecords[ResponsiblePeople](c1, ResponsiblePeople.key)
           } yield c2
 
           //there is data in S4l
-          fix flatMap { cacheMap =>
+          fixEmpties flatMap { cacheMap =>
             if (dataHasChanged(cacheMap)) {
-              (cacheMap.getEntry[SubscriptionResponse](SubscriptionResponse.key),
-                cacheMap.getEntry[AmendVariationRenewalResponse](AmendVariationRenewalResponse.key)) match {
-                case (Some(_), _) => refreshAndRedirect(amlsRegistrationNumber, Some(cacheMap))
-                case (_, Some(_)) => refreshAndRedirect(amlsRegistrationNumber, Some(cacheMap))
-                case _ => setAlCorrespondenceAddressAndRedirect(amlsRegistrationNumber, Some(cacheMap))
-
+              cacheMap.getEntry[SubmissionRequestStatus](SubmissionRequestStatus.key) collect {
+                case SubmissionRequestStatus(true) => refreshAndRedirect(amlsRegistrationNumber, Some(cacheMap))
```

Comment:
Have we lost the AmendVariationRenewalResponse????

Reply:
The original logic here relied on either `SubscriptionResponse` or `AmendVariationRenewalResponse` data being in the cache in order to be able to tell us whether or not the user has just done a submission. Originally, this data was not retained when the cache was cleared and so this logic is fine.

Now though, this data _is_ being retained when the cache is cleared and so this logic here is no longer a reliable indicator as to whether the user has just submitted an application or not. Thus, that has been replaced by a new type `SubmissionRequestStatus` which is trashed when the data cache is cleared immediately following a submission.

Happy to discuss.

Reply:
Happy to merge. There's a tiny thing we can do in the messages file concerning passing the link into it but it's very minor and I'll pick that up with you later.